pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -174,6 +174,9 @@ public static List<String> extractTableNamesFromNode(SqlNode sqlNode) {
       }
       tableNames.addAll(extractTableNamesFromNode(((SqlWith) sqlNode).body));
       tableNames.removeAll(aliases);
+    } else if (sqlNode instanceof SqlExplain) {
+      SqlExplain explain = (SqlExplain) sqlNode;
+      tableNames.addAll(extractTableNamesFromNode(explain.getExplicandum()));
```

Comment:
This was never added previously. how come this become a problem after the broker tenant changes?

Reply:
This was a miss in the previous PR. In case of multistage queries via controller, we extract the table name from the query to see all the tables are in the same tenant or not. In case of `explain plan for ...` queries, we missed out on extracting name in the parser. Added it here in this patch.